ui/components/app/snaps/snap-permissions-list/snap-permissions-list.test.js

```diff
@@ -58,9 +58,13 @@ describe('Snap Permission List', () => {
       />,
       store,
     );
-    expect(
-      screen.getByText('Display dialog windows in MetaMask.'),
-    ).toBeInTheDocument();
-    expect(screen.getByText('Approved on 2023-04-05')).toBeInTheDocument();
+    waitFor(() => {
```

Comment:
Another argument as to not storing the permissions in state is having to write tests like this

Reply:
Updated implementation with new approach.